presto-hive/src/main/java/io/prestosql/plugin/hive/rubix/RubixInitializer.java

```diff
@@ -54,39 +55,44 @@
     private static final Logger log = Logger.get(RubixInitializer.class);
 
     private final RetryPolicy retryPolicy;
+    private final NodeManager nodeManager;
     private final CatalogName catalogName;
     private final RubixConfigurationInitializer rubixConfigurationInitializer;
     private final HdfsConfigurationInitializer hdfsConfigurationInitializer;
 
     @Inject
     public RubixInitializer(
+            NodeManager nodeManager,
             CatalogName catalogName,
             RubixConfigurationInitializer rubixConfigurationInitializer,
             HdfsConfigurationInitializer hdfsConfigurationInitializer)
     {
-        this(RETRY_POLICY, catalogName, rubixConfigurationInitializer, hdfsConfigurationInitializer);
+        this(RETRY_POLICY, nodeManager, catalogName, rubixConfigurationInitializer, hdfsConfigurationInitializer);
     }
 
     @VisibleForTesting
     RubixInitializer(
             RetryPolicy retryPolicy,
+            NodeManager nodeManager,
             CatalogName catalogName,
             RubixConfigurationInitializer rubixConfigurationInitializer,
             HdfsConfigurationInitializer hdfsConfigurationInitializer)
     {
         this.retryPolicy = retryPolicy;
+        this.nodeManager = nodeManager;
         this.catalogName = catalogName;
         this.rubixConfigurationInitializer = rubixConfigurationInitializer;
         this.hdfsConfigurationInitializer = hdfsConfigurationInitializer;
     }
 
-    public void initializeRubix(NodeManager nodeManager)
+    @PostConstruct
```

Comment:
Did you test if breaks presto startup in case of failure with `@PostConstruct`. My understanding is it should but please verify.